app/models/ems_refresh/manager.rb

```diff
@@ -4,6 +4,7 @@ module Manager
     module ClassMethods
       unless respond_to?(:ems_type)
         def ems_type
+          ActiveSupport::Deprecation.warn("ems_type is required, please ensure you're using a leaf class.", caller[1..-1])
```

Comment:
...please ensure you're using a leaf class that implements it.

Reply:
👅

Reply:
Also the ```caller[1..-1]``` is basically useless here since all it does is tell you in the majority of cases that ```/home/travis/build/ManageIQ/manageiq/spec/support/missing_factory_helper.rb:10``` is the culprit

Reply:
And don't all leaf classes if they're true leaves implement it? Also @jrafanie yeah, I know about the other repos, about 90 UI classic specs fail with this